rct229/rulesets/ashrae9012019/section19/section19rule7.py

```diff
@@ -0,0 +1,342 @@
+        for space_b in find_all(
+            "$.buildings[*].building_segments[*].zones[*].spaces[*]",
+            rmi_b,
+        ):
+            if hvac_system_serves_only_labs:
+                lighting_space_type_b = space_b.get("lighting_space_type")
+                if lighting_space_type_b is None:
+                    all_lighting_space_types_defined = False
+                else:
+                    are_any_lighting_space_types_defined = True
+                    if (
+                        lighting_space_type_b
+                        != LIGHTING_SPACE.LABORATORY_EXCEPT_IN_OR_AS_A_CLASSROOM
+                    ):
+                        hvac_system_serves_only_labs = False
+            else:
+                all_lighting_space_types_defined = False
+
+        zone_data = {}
+        for hvac_id_b in find_all(
+            "$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*].id",
+            rmi_b,
+        ):
+            zone_data[hvac_id_b] = {
+                "zone_OA_CFM_list_of_schedules_b": [],
+                "zone_OA_CFM_list_of_schedules_p": [],
+                "was_DCV_modeled_baseline": False,
+                "was_DCV_modeled_proposed": False,
+                "zone_air_distribution_effectiveness_greater_than_1": False,
+            }
+            for zone_id_b in dict_of_zones_and_terminal_units_served_by_hvac_sys_b[
+                hvac_id_b
+            ]["zone_list"]:
+                zone_data[hvac_id_b]["zone_OA_CFM_list_of_schedules_b"].append(
+                    get_min_oa_cfm_sch_zone(rmi_b, zone_id_b)
+                )
+                zone_data[hvac_id_b]["zone_OA_CFM_list_of_schedules_p"].append(
+                    get_min_oa_cfm_sch_zone(rmi_p, zone_id_b)
+                )
+
+                zone_data[hvac_id_b]["was_DCV_modeled_baseline"] = any(
+                    [
+                        True
+                        if getattr_(
+                            terminal_b, "Terminal", "has_demand_control_ventilation"
+                        )
+                        else False
+                        for terminal_b in find_all(
+                            f'$.buildings[*].building_segments[*].zones[*][?(@.id = "{zone_id_b}")].terminals[*]',
+                            rmi_b,
+                        )
+                    ]
+                )
+
+                zone_p = find_one(
+                    f'$.buildings[*].building_segments[*].zones[*][?(@.id = "{zone_id_b}")]',
+                    rmi_p,
+                )
+
+                zone_data[hvac_id_b]["was_DCV_modeled_proposed"] = (
+                    find_one(
+                        f'$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*][?(@.id = "{hvac_id_b}")].fan_system.demand_control_ventilation_control',
+                        rmi_p,
+                    )
+                    != DEMAND_CONTROL_VENTILATION_CONTROL.NONE
+                )
```

Comment:
You will also need to add Null to the case.
`find_one(...) not in [None, DEMAND_CONTROL_VENTILATION_CONTROL.NONE]`

This logic seems incorrect. I think you will always getting None from the `find_one` function because the function is looking for `hvac_id_b` in `rmi_p`.

In RDS, it says:
                - Check if the hvac system serving the terminal unit has dcv modeled (weili may have to include some logic to get the hvac object from the id): `if terminal.served_by_heating_ventilating_air_conditioning_system.fan_system.demand_control_ventilation_control == true: was_DCV_modeled_proposed`

Reply:
Great point! Addressed.